### PR TITLE
Fix voting notification bug

### PR DESCRIPTION
**Pull request by Dominik-Weinzierl**
<h2> Description </h2>

Instead of displaying the start of the voting phase (or other voting notifications) only when the voting phase started, it was displayed for every board edit event.

## Changelog 
- Add undefined check for `action.board.voting` in `boardReducer`

### CODE REVIEW

File: src/store/reducer/board.ts [hunk 1]
```diff
@@ -13,8 +13,8 @@ export const boardReducer = (state: BoardState = {status: "unknown"}, action: Re
       };
     }
     case ActionType.EditBoard: {
-      // Moderator started voting phase - notification to moderator (user who started the voting)
-      if (state.data?.voting !== action.board.voting) {
+      // Moderator started voting phase - notification to moderator (user w″ho started the voting)
```
Comment (brandstetterm) [inline src/store/reducer/board.ts:16]: Typo

File: src/store/reducer/board.ts [hunk 2]
```diff
@@ -13,8 +13,8 @@ export const boardReducer = (state: BoardState = {status: "unknown"}, action: Re
...
+      if (state.data?.voting !== action.board.voting && action.board.voting) {
```
Comment (brandstetterm) [inline src/store/reducer/board.ts:17]: In theory `if (action.board.voting) {...}` should be enough. `action.board.voting` is undefined if it's not getting changed

File: src/store/reducer/board.ts [hunk 3]
```diff
@@ -86,7 +86,7 @@ export const boardReducer = (state: BoardState = {status: "unknown"}, action: Re
     }
     case ActionType.UpdatedBoard: {
       // User notification
-      if (state.data?.voting !== action.board.voting) {
+      if (state.data?.voting !== action.board.voting && action.board.voting) {
```
Comment (brandstetterm) [inline src/store/reducer/board.ts:89]: `action.board.voting` is always defined

File: src/store/reducer/board.ts (hunk 1 above)
Comment (Dominik-Weinzierl) [reply src/store/reducer/board.ts:16]: Thank you

File: src/store/reducer/board.ts (hunk 3 above)
Comment (Dominik-Weinzierl) [reply src/store/reducer/board.ts:89]: True

File: src/store/reducer/board.ts (hunk 2 above)
Comment (Dominik-Weinzierl) [reply src/store/reducer/board.ts:17]: Yes, that works

File: src/store/reducer/board.ts [hunk 4]
```diff
@@ -13,13 +13,11 @@ export const boardReducer = (state: BoardState = {status: "unknown"}, action: Re
...
     }
     case ActionType.EditBoard: {
-      // Moderator started voting phase - notification to moderator (user who started the voting)
-      if (state.data?.voting !== action.board.voting) {
-        if (action.board.voting === "active") {
-          Toast.success("You started the voting phase!");
-        } else {
-          Toast.error("You ended the voting phase!");
-        }
+      // Moderator started voting phase - notifica″tion to moderator (user who started the voting)
```
Comment (brandstetterm) [inline src/store/reducer/board.ts:16]:
```suggestion
      // Moderator started voting phase - notification to moderator (user who started the voting)
```

File: src/store/reducer/board.ts [hunk 5]
```diff
@@ -13,13 +13,11 @@ export const boardReducer = (state: BoardState = {status: "unknown"}, action: Re
...
+      if (action.board.voting === "active") {
+        Toast.success("You started the voting phase!");
+      } else if (action.board.voting === "disabled") {
+        Toast.error("You ended the voting phase!");
```
Comment (brandstetterm) [inline src/store/reducer/board.ts:20]: Wie wärs damit? :P

```suggestion
      if (action.board.voting) {
        Toast.success(`You ${action.board.voting === "active" ? "started" : "ended"} the voting phase!`);
      } 
```

File: src/store/reducer/board.ts (hunk 4 above)
Comment (Dominik-Weinzierl) [reply src/store/reducer/board.ts:16]: Interesting. Thank you.

File: src/store/reducer/board.ts (hunk 5 above)
Comment (Dominik-Weinzierl) [reply src/store/reducer/board.ts:20]: :P

**Review by brandstetterm — APPROVED** [review]

**Pull request merged**